clang/test/SemaHLSL/resource_binding_attr_error_udt.hlsl

```diff
@@ -0,0 +1,104 @@
+// RUN: %clang_cc1 -triple dxil-pc-shadermodel6.3-library -x hlsl -o - -fsyntax-only %s -verify
+
+// TODO: Implement "Buffer"
+struct Eg1 {
+  float f;
+  // Buffer<float> Buf;

```

Comment:
This file has a bunch of members and binding annotations commented out.  Is that intentional?

Reply:
Yes, it is intentional. The commented-out examples cannot be run in any valuable sense, without some prerequisite implementation of types that are not yet implemented.
For example, after `Buffer` is implemented, or after `Texture2D` is implemented, we can better test the diagnostic infrastructure. We don't have any resources that can bind to the `t` register type yet, so tests are necessarily quite limited.

Reply:
I guess we don't have the ability to create resource types using just the attributes that we can use in the tests, like how Helena did for the availability diagnostics?

Reply:
No, it isn't possible, because the tests in udt depend on both the declaration of the variable and the definition of the variable type. Even when adding the availability attribute, the `Texture2D` token just isn't parseable.

Reply:
It's a shame that there's no way to spell HLSLResource :(